data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,441 @@
+                        clean_consts.DESTINATION_DATASET: date_row.dataset_id,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
+                    })
+                else:
+                    sandbox_query = SANDBOX_QUERY_END_DATE.format(
+                        project=date_row.project_id,
+                        sandbox_dataset=sandbox_dataset,
+                        intermediary_table=ticket_number + '_' + date_row.table,
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: sandbox_query,
+                        clean_consts.DESTINATION_DATASET: sandbox_dataset,
+                        clean_consts.DESTINATION_TABLE: date_row.table
+                    })
+                    clean_query = CLEAN_QUERY_END_DATE.format(
+                        project=date_row.project_id,
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: clean_query,
+                        clean_consts.DESTINATION_DATASET: date_row.dataset_id,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE
+                    })
+            else:
+                # break out of loop to create query, if pid does not exist in table
+                continue
+    print(
+        'Query list complete, retracting ehr deactivated PIDS from the following datasets: %s'
+        % date_columns_df['dataset_id'].tolist())
+    logging.info(
+        'Query list complete, retracting ehr deactivated PIDS from the following datasets: %s'
+        % date_columns_df['dataset_id'].tolist())
+    return queries_list
+
+
+def run_queries(queries):
+    """
+    Function that will perform the retraction.
+
+    :param queries: list of queries to run retraction
+    """
+    query_job_ids = []
+    for query_dict in queries:
+        if 'sandbox' in query_dict['destination_dataset_id']:
```

Comment:
Right now, this business logic will work.  But, if we ever changed the naming convention of the sandbox datasets this condition could be impacted.  A suggestion to think about, would be to add a key/value pair to your query dictionary to indicate what type of query you are running.  You should be able to easily add this when you are generating your query.  It could be something like `{'type': 'sandbox' or 'retraction'}`.  Then you could check this and it would be guaranteed to work even if dataset names change.  Thoughts/concerns/suggestions?

Reply:
per our conversation, totally agree. Adding a key/value to query then will check `if query_dict['type'] == 'sandbox':`